core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -1064,6 +1021,24 @@ public void shutdown() {
         result.mergeWith(dataAccess.deleteMetricData(id).flatMap(r -> null));
 
         result.mergeWith(dataAccess.deleteMetricFromRetentionIndex(id).flatMap(r -> null));
+        result.mergeWith(dataAccess.deleteFromMetricExpirationIndex(id).flatMap(r -> null));
+
         return result;
     }
+
+    @Override
+    public <T> void updateMetricExpiration(Metric<T> metric) {
+        if (!MetricType.STRING.equals(metric.getType())) {
+            long expiration = 0;
```

Comment:
I would not do any extra work because that will just change the time by an hour or two, since the job is run in daily increments it will not make much of a difference.